File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.js
function solution(s) {
  let ans = [];
  const testCase = s.split(" ");
  for (const item of testCase) {
    if (item) {
      // 첫 문자가 알파벳인 경우
      if (isNaN(item[0])) {
        const addItem = item.toLowerCase().split("");
        addItem[0] = addItem[0].toUpperCase();
        ans.push(addItem.join(""));
        // 첫 문자가 알파벳이 아닌 경우
      } else {
        ans.push(item.toLowerCase());
      }
    } else {
      ans.push(item);
    }
  }
  return ans.join(" ");
}

File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = []
    s = s.split(" ")
    
    for item in s:
        if len(item):
            # 숫자가 아닌 경우
            if not item[0].isdigit():
                newItem = item[0].upper() + item[1:].lower()
                answer.append(newItem)
            else:
            # 숫자인 경우
                answer.append(item.lower())
        else:
            answer.append("")
    print(answer)
    return " ".join(answer)